<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SerialPortDownloaderClass</class>
 <widget class="QMainWindow" name="SerialPortDownloaderClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>473</width>
    <height>420</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <family>黑体</family>
    <pointsize>14</pointsize>
    <bold>false</bold>
   </font>
  </property>
  <property name="windowTitle">
   <string>Serial Port Downloader</string>
  </property>
  <property name="windowIcon">
   <iconset resource="SerialPortDownloader.qrc">
    <normaloff>:/Icon/resource/cxk_icon.ico</normaloff>:/Icon/resource/cxk_icon.ico</iconset>
  </property>
  <property name="styleSheet">
   <string notr="true">QLabel, QAbstractButton {
    font: bold;
	font: 14pt &quot;黑体&quot;;
}

/* 未下拉时，QComboBox的样式 */
QComboBox {
    font: bold;
	font: 14pt &quot;黑体&quot;;
    border-radius: 3px;
    padding: 1px 18px 1px 3px;
    background: transparent; 
    border: 1px solid gray; 
	color: #333333;
	border-color: #E5E5E5;
	background-color: #FFFFFF;
}
/* 点击QComboBox后的已选中项的样式 */
QComboBox:on {
    color: rgb(33, 188, 188);
    background-color: #f1ee04;
}

/* 下拉展开后，整个下拉窗体样式 */
QComboBox QAbstractItemView {
    outline: 1px solid #000000;/*选中项外边框*/
    border: 1px solid yellow;/* 整个下拉窗体的边框 */
    color: #d34b4b;
    background-color: #FFFFFF; /* 整个下拉窗体的背景色 */
    selection-color: #3377FF;/*下拉框选中项字体颜色*/
    selection-background-color:#FFFFFF;/* 下拉框选中项的背景色 */
}

QPushButton
{
	background-color: #473C8B; /*背景色*/ 
	border-style: outset;
	border-width: 2px;
	border-radius:5px; /*边界圆滑*/
	border-color: #8B7355;
	font: bold 18px;
	min-width:2em;
	color:white; /*字体颜色*/
	font-family:华文新魏;
	padding: 5px;
}
 
QPushButton:hover
{
	background-color: rgb(0, 150, 0);
}
 
QPushButton:pressed
{
	background-color: #1E90FF; /*伪状态经过时背景色*/ 
	border-style: inset;
}
QPushButton:!enabled{
	background-color: rgb(100, 100, 100);
	border-style: inset;
}</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QLabel" name="curPortLabel">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>60</y>
      <width>91</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>当前串口：</string>
    </property>
   </widget>
   <widget class="QComboBox" name="portComboBox">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>70</y>
      <width>101</width>
      <height>22</height>
     </rect>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="baudRateLabel">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>30</y>
      <width>71</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>波特率：</string>
    </property>
   </widget>
   <widget class="QComboBox" name="baudRateComboBox">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>40</y>
      <width>101</width>
      <height>22</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>3</number>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
    <item>
     <property name="text">
      <string>9600</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>14400</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>19200</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>115200</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="portScanButton">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>100</y>
      <width>111</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>扫描串口</string>
    </property>
   </widget>
   <widget class="QPushButton" name="portSendButton">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>320</y>
      <width>81</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>发送</string>
    </property>
   </widget>
   <widget class="QPushButton" name="selectImageButton">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>210</y>
      <width>111</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>选择图片</string>
    </property>
   </widget>
   <widget class="QPushButton" name="openPortButton">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>150</y>
      <width>111</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>打开串口</string>
    </property>
   </widget>
   <widget class="QLabel" name="previewLabel">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>40</y>
      <width>200</width>
      <height>200</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="previewTextLabel">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>10</y>
      <width>51</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>预览</string>
    </property>
   </widget>
   <widget class="QPushButton" name="binarizeButton">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>270</y>
      <width>81</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>二值化</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="plainTextEdit">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>260</y>
      <width>201</width>
      <height>141</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>黑体</family>
      <pointsize>18</pointsize>
      <bold>false</bold>
     </font>
    </property>
    <property name="plainText">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="syncButton">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>270</y>
      <width>71</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>同步</string>
    </property>
   </widget>
   <widget class="QPushButton" name="clearButton">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>320</y>
      <width>71</width>
      <height>41</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="text">
     <string>清空</string>
    </property>
   </widget>
   <widget class="QComboBox" name="binarizationComboBox">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>380</y>
      <width>183</width>
      <height>25</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>黑体</family>
      <pointsize>14</pointsize>
      <italic>false</italic>
      <bold>false</bold>
      <kerning>true</kerning>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="sizeAdjustPolicy">
     <enum>QComboBox::AdjustToContents</enum>
    </property>
    <item>
     <property name="text">
      <string>全局阈值</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>自适应阈值</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Floyd Steinberg</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>JJN Dithering</string>
     </property>
    </item>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>baudRateComboBox</tabstop>
  <tabstop>portComboBox</tabstop>
  <tabstop>portScanButton</tabstop>
  <tabstop>openPortButton</tabstop>
  <tabstop>selectImageButton</tabstop>
  <tabstop>binarizeButton</tabstop>
  <tabstop>portSendButton</tabstop>
  <tabstop>syncButton</tabstop>
  <tabstop>clearButton</tabstop>
  <tabstop>plainTextEdit</tabstop>
 </tabstops>
 <resources>
  <include location="SerialPortDownloader.qrc"/>
 </resources>
 <connections/>
</ui>
